src/lib/es2015.core.d.ts

```diff
@@ -10,6 +10,9 @@ interface Array<T> {
      * @param thisArg If provided, it will be used as the this value for each invocation of
      * predicate. If it is not provided, undefined is used instead.
      */
+    find<S extends T>(predicate: (this: void, value: T, index: number, obj: Array<T>) => value is S): S | undefined;
+    find<S extends T>(predicate: (this: void, value: T, index: number, obj: Array<T>) => value is S, thisArg: undefined): S | undefined;
+    find<S extends T, Z>(predicate: (this: Z, value: T, index: number, obj: Array<T>) => value is S, thisArg: Z): S | undefined;

```

Comment:
At a glance, it seems like the first two signatures aren't needed if the thisArg parameter is made optional in the third.